src/theme/src/hooks/useCheckboxAppearance.js

```diff
@@ -1,140 +1,35 @@
-    disabled: disabledStyles = {},
-    hover: hoverStyles = {},
-    focus: focusStyles = {},
-    active: activeStyles = {},
-    checked: checkedStyles = {},
-    checkedHover: checkedHoverStyles = {},
-    checkedDisabled: checkedDisabledStyles = {},
-    checkedActive: checkedActiveStyles = {}
-  } = checkbox || {}
-
-  return {
-    ...hiddenCheckboxStyle,
-    '& + div > svg': { display: 'none' },
-    [defaultState]: {
-      ...baseStyle,
-      color: 'white',
-      backgroundColor: 'white',
-      boxShadow: `inset 0 0 0 1px ${colors.gray400}`,
-      ...baseStyles
-    },
-    [disabledState]: {
-      cursor: 'not-allowed',
-      backgroundColor: colors.gray100,
-      boxShadow: `inset 0 0 0 1px ${colors.gray100}`,
-      ...disabledStyles
-    },
-    [hoverState]: {
-      boxShadow: `inset 0 0 0 1px ${colors.gray600}`,
-      ...hoverStyles
-    },
-    [focusState]: {
-      boxShadow: `0 0 0 2px ${colors.blue100}`,
-      ...focusStyles
-    },
-    [activeState]: {
-      backgroundColor: colors.gray100,
-      boxShadow: `inset 0 0 0 1px ${colors.gray500}`,
-      ...activeStyles
-    },
-    [checkedState]: {
-      ...checkedStyle,
-      color: 'white',
-      boxShadow: `inset 0 0 0 -1px ${primary.active}`,
-      backgroundColor: primary.base,
-      ...checkedStyles
-    },
-    [checkedHoverState]: {
-      ...checkedStyle,
-      color: 'white',
-      backgroundColor: primary.hover,
-      boxShadow: `inset 0 0 0 1px ${primary.hover}`,
-      ...checkedHoverStyles
-    },
-    [checkedDisabledState]: {
-      ...checkedStyle,
-      color: colors.gray600,
-      backgroundColor: colors.gray100,
-      ...checkedDisabledStyles
-    },
-    [checkedActiveState]: {
-      ...checkedStyle,
-      color: 'white',
-      boxShadow: `inset 0 0 0 -1px ${primary.active}`,
-      backgroundColor: primary.active,
-      ...checkedActiveStyles
-    }
-  }
+const pseudoSelectors = {
+  _display: '& + div',
+  _tickSvg: '& + div > svg',
+  _disabled: '&:disabled + div',
+  _hover: '&:not(:disabled):hover + div',
+  _focus: '&:not(:disabled):focus + div',
+  _active: '&:not(:disabled):active + div',
+  _checked: '&:checked + div, &[type=checkbox]:indeterminate + div',
+  _checkedHover: '&:not([disabled]):checked:hover + div, &[type=checkbox]:not([disabled]):indeterminate:hover + div',
+  _checkedActive: '&:not([disabled]):checked:active + div, &[type=checkbox]:not([disabled]):indeterminate:active + div',
+  _checkedDisabled: '&[disabled]:checked + div, &[type=checkbox][disabled]:indeterminate + div'
 }
 
 /**
  * Get the className of a `Checkbox`.
  * @param {string} appearance
  * @return {string} the appearance class name.
  */
-function useCheckboxAppearance() {
+function useCheckboxAppearance(modifiers) {
   const theme = useTheme()
-  const className = useMemo(() => css(getCheckboxStyles(theme)).toString(), [
-    theme
-  ])
-  return className
+  const checkboxStyles = useMemo(() => getCheckboxStyles(theme), [theme])
```

Comment:
The thing that I'm _slowly_ starting to come around on a full theme implementation is that we need some way for _most_ of the visual styles to be derived from the theme in some way (along with some of the basic box props). 

The previous `useCheckboxAppearance` was an early-ish experiment to see what overriding styles could look like, but with this "style config" approach, I think that we can push a lot of the "visual" styles for the checkbox (i.e. checked background color, border, box shadow) into the theme, while leaving a lot of the internal "reset" styles to be ...well, internal. What do you think?